qdrant-landing/content/articles/solving-contexto.md

```diff
@@ -0,0 +1,177 @@
+
+{{< figure src=/articles_data/solving-contexto/cbow-word2vec.webp caption="Word2Vec training architecture">}}
+
+
+Word2vec is one of the first methods used to represent objects in vector space. 
+Currently there are a lot of more sophisticated methods, that can capture meaning of the whole texts, not just a single word.
+But for our purposes word2vec will work just fine.
+
+So, here's the naive approach you've probably already thought of:
+
+We can start with a random word and look into the list of similar words using some Word2Vec model.
+If we see a word closer to the secret word, we use it as a reference and repeat the process.
+
+Although this approach works if we are initially close enough to the secret word, it is generally quite slow and inefficient.
+It tends to get stuck in clusters of words that are similar to each other, forcing us to retrieve many words from the model.
+
+Additionally, using linear algebra techniques to evaluate the exact vector based on distances to given points does not look feasible in this scenario.
+This is because the exact word2vec model used to sort the words is unknown, as is the exact distance to the secret word.
+The only option is to compare distances between words.
+
+
+## One working approach
+
+Based on the previous section, we can conclude, that using only the most similar word found so far is not enough to generate efficient guesses. 
+The more efficient solution must also consider the words deemed dissimilar.
+
+Let's consider the simplest case: we have guessed 2 words `house` and `blue` and received feedback on their similarity to the secret word.
+
+One of the words is closer to the secret word than the other, so we can make some assumptions about the secret word.
+We understand that the secret word is more likely to be similar to `house` than `blue`, but we only have the informaion about its relative similarity to these two words.
+
+Let's assign a score to each word in the vocabulary based on this observation:
+
+{{< figure src=/articles_data/solving-contexto/scoring-1.png caption="Scoring words based on 2 guesses">}}
+
+We assign +1 score to those words that are closer to `house` than `blue` and -1 score to those words that are closer to `blue` than `house`.
+
+Now, we can use this score to rank the words in the vocabulary and use word with the highest score as our next guess.
+
+Let's see how scores change after we make a third guess:
+
+{{< figure src=/articles_data/solving-contexto/scoring-2.png caption="Ranking words based on next 2 guesses">}}
+
+We can generalize this approach to any number of guesses.
+The simpliest way to do this is to sample pairs of guesses and update the score iteratively.
+
+That's it! We can use this approach to suggest words one by one and extend guess list accordingly.
```

Comment:
I actually expected the approach would be to use the recommendation API, or even just search with all the embeddings stored in Qdrant. That might be an interesting case to promote, but there is no reference to Qdrant, while there could be.

Reply:
there is reference in the last section